difflint/lib.py

```diff
@@ -2,24 +2,44 @@
 from datetime import datetime
 from difflib import unified_diff
 from io import StringIO
-from os import unlink
+from os import getcwd, unlink
+import os.path
+import pathlib
 import re
-from subprocess import call, check_output
+from subprocess import call, CalledProcessError, check_output, DEVNULL
 from sys import stderr
 
 from .lint import get_missing_configuration_files, get_missing_linters, lint
 
 LOG_FILE = 'lintdiff.log'
 MISSING_FILE_EXIT_CODE = 72  # os.EX_OSFILE is not portable
 
+def repo_root():
+    """Returns the root of the current repository. This is the directory
+    containing the .git directory for this repository.
+
+    Input: None
+
+    Output: A pathlib.Path object representing the root of this repository.
+    """
+    directory = pathlib.Path.cwd()
+    while directory.parent != directory.root:
```

Comment:
Consider iterating over https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.parents to make this even simpler.